### ISSUE TITLE
sqlite3.ProgrammingError  SQLite objects created in a thread can only be used in that same thread.

### ISSUE THREAD

**Issue by rth** [icubam/icubam#321]
There were a few error reports with the following error,
```
sqlite3.ProgrammingError  SQLite objects created in a thread can only be used in that same thread.
```

while doing `do_close` or `do_rolback` in `sqlalchemy/engine/default.py`.

As far as I can tell, the most likely cause is some bad interaction between threaded NGNIX workers and sqlalchemy. Needs more investigation.

**Comment by dulacarnold** [comment]
Yeah, this is a tough one, we had a bug like this a week or two ago and we never quite figured it out (ended up taking another approach).  Unfortunately Sentry doesn't really tell us /where/ the problem is...
